rollastic/node.py

```diff
@@ -54,13 +55,26 @@ def populate(self):
         self.clear()
 
         info = self.cluster.es.nodes.info(self.node_id)['nodes'].get(self.node_id, {})
+        isbad = False
         if not info:
-            _LOG.warning('Bad result for node info. node_id=%s info=%s', self.node_id, info)
+            msg = "Bad result for node info. node_id={0} info={1}".format(self.node_id, info)
+            _LOG.warning(msg)
+            isbad = True
         self.update(info)
 
         stats = self.cluster.es.nodes.stats(self.node_id)['nodes'].get(self.node_id, {})
         if not stats:
-            _LOG.warning('Bad result for node stats. node_id=%s stats=%s', self.node_id, stats)
+            msg = "Bad result for node stats. node_id={0} stats={1}".format(self.node_id, stats)
+            _LOG.warning(msg)
+            isbad = True
+
+        # Make sure that we don't infintely wait on nodes that can't get a good result
+        if isbad:
+            self.badcount += 1
+            if self.badcount > 10:
```

Comment:
I've modified the class constructor to take an optional parameter for this, so I can explore making this configurable.